wrangler-core/src/main/java/io/cdap/wrangler/utils/ColumnConverter.java

```diff
@@ -210,25 +213,30 @@ private static Object convertType(String col, String toType, Object object) thro
       }
 
       case "DECIMAL": {
+        BigDecimal decimal = null;
         if (object instanceof BigDecimal) {
-          return object;
+          decimal = (BigDecimal) object;
         } else if (object instanceof String) {
-          return new BigDecimal((String) object);
+          decimal = new BigDecimal((String) object);
         } else if (object instanceof Integer) {
-          return BigDecimal.valueOf((Integer) object);
+          decimal = BigDecimal.valueOf((Integer) object);
         } else if (object instanceof Short) {
-          return BigDecimal.valueOf((Short) object);
+          decimal = BigDecimal.valueOf((Short) object);
         } else if (object instanceof Long) {
-          return BigDecimal.valueOf((Long) object);
+          decimal = BigDecimal.valueOf((Long) object);
         } else if (object instanceof Float) {
-          return BigDecimal.valueOf((Float) object);
+          decimal = BigDecimal.valueOf((Float) object);
         } else if (object instanceof Double) {
-          return BigDecimal.valueOf((Double) object);
+          decimal = BigDecimal.valueOf((Double) object);
         } else if (object instanceof byte[]) {
-          return Bytes.toBigDecimal((byte[]) object);
-        } else {
+          decimal = Bytes.toBigDecimal((byte[]) object);
+        }
+        if (decimal == null) {
           break;
+        } else if (roundingMode.equals(RoundingMode.UNNECESSARY) && (decimal.scale() != scale)) {
+          throw new DirectiveExecutionException("Input value's scale is not equal to set output schema scale");
```

Comment:
Why is this line needed? Flow from UI(not command line), these new args wont be set and do we throw an exception in that case?